### PR TITLE
deal with readings having no sign display user

### PR DESCRIPTION
**Pull request by yCobanoglu**

### CODE REVIEW

File: src/signs/infrastructure/SignRepository.ts [hunk 1]
```diff
@@ -3,42 +3,54 @@ import Promise from 'bluebird'
...
 import { AnnotationToken } from 'fragmentarium/ui/image-annotation/annotation-tool/annotation-token'
+import { AnnotationTokenType } from 'fragmentarium/domain/annotation'
 
 class SignRepository {
   private readonly apiClient
 
   constructor(apiClient: ApiClient) {
     this.apiClient = apiClient
   }
+  private attachSignToToken(
```
[the remark this answers is not in the document]
Comment (jlaasonen) [reply src/signs/infrastructure/SignRepository.ts:29]: You could extract constant/function for the check. Having a name for the condition makes it easier to understand and you can more convenient use the ternary operator.

File: src/test-support/lines/text-line.ts [hunk 2]
```diff
@@ -56,13 +56,13 @@ export default new TextLine({
...
         {
-          enclosureType: ['BROKEN_AWAY', 'PERHAPS_BROKEN_AWAY'],
+          enclosureType: [],
           cleanValue: 'ŠA₂',
           value: 'ŠA₂',
           name: 'ŠA',
           nameParts: [
             {
-              enclosureType: ['BROKEN_AWAY', 'PERHAPS_BROKEN_AWAY'],
+              enclosureType: [],
```
Comment (jlaasonen) [inline src/test-support/lines/text-line.ts:65]: Why are these removed?

**Review by jlaasonen — CHANGES_REQUESTED** [review]

File: src/fragmentarium/ui/display/__snapshots__/Display.test.tsx.snap [hunk 3]
```diff
@@ -248,13 +248,13 @@ exports[`Snapshot 1`] = `
...
                   <span
-                    class="Transliteration__Logogram--BROKEN_AWAY Transliteration__Logogram--PERHAPS_BROKEN_AWAY"
+                    class=""
                   >
                     <span
                       class="Transliteration__ValueToken"
                     >
                       <span
-                        class="Transliteration__ValueToken--BROKEN_AWAY Transliteration__ValueToken--PERHAPS_BROKEN_AWAY"
+                        class=""
```
[the remark this answers is not in the document]
Comment (jlaasonen) [reply src/fragmentarium/ui/display/__snapshots__/Display.test.tsx.snap:257]: Enclosures were removed from the test data.

File: src/test-support/lines/text-line.ts (hunk 2 above)
Comment (yCobanoglu) [reply src/test-support/lines/text-line.ts:65]: i use them in the FragmentAnnotation.test.tsx where they wouldn't be displayed if they are BROKEN so they need not to be broken.

File: src/fragmentarium/ui/image-annotation/annotation-tool/annotation-token.ts [hunk 4]
```diff
@@ -57,76 +90,132 @@ export class AnnotationToken {
...
+            : AnnotationTokenType.HasSign,
+          token.value,
+          path,
+          namedSign.name,
+          namedSign.subIndex
+        )
+      }
+    }
+    default: {
+      const compoundGrapheme = token as CompoundGrapheme
```
[the remark this answers is not in the document]
Comment (yCobanoglu) [reply src/fragmentarium/ui/image-annotation/annotation-tool/annotation-token.ts:133]: this line: `if (
    ['Reading', 'Logogram', 'CompoundGrapheme', 'Number'].includes(token.type)
  )`.
  If it's none the the first three which are checked right after than it has to be a CompoundGrapheme.

File: src/test-support/lines/text-line.ts (hunk 2 above)
Comment (jlaasonen) [reply src/test-support/lines/text-line.ts:65]: Do not break existing tests. The new enclosure type is incorrect given the enclosures in the line.

If you need something else create new data. You can add factories to reduce duplication.

Comment (yCobanoglu) [reply src/test-support/lines/text-line.ts:65]: ok i added a reading to textline. All tests pass.

**Review by jlaasonen — APPROVED** [review]
